src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1315,6 +1339,24 @@ internal string GetTablePrefix(string databaseName, string schemaName)
                     {
                         columnDefinition.DefaultValue = columnInfo["COLUMN_DEFAULT"];
                     }
+
+                    columnDefinition.DbType = TypeHelper.GetDbTypeFromSystemType(columnDefinition.SystemType);
+                    if (GetDatabaseType() is DatabaseType.MSSQL &&
```

Comment:
Same comment - why do specific things for MsSql in a base class? Can we have an overridden derived class implementation?

Reply:
Sure, we can have that. But then we cannot throw exception in the default implementation of the method and would have to instead return null.

Reply:
> Sure, we can have that. But then we cannot throw exception in the default implementation of the method and would have to instead return null. Since currently we are throwing an exception in the default implementation of the method, we need to ensure that we only call the method for MsSql and hence this check at L1344.

Can you explain more about why this can not remain void if we have the different behavior for MsSql split into an overriding function in the MsSql class? I am not following why we cant just have the MsSql specific code attached to a function in MsSql that overrides this and calls a shared helper that executes the code shared between them.

Reply:
Keeping the function void would require to write two different functions for ParameterDefinition and ColumnDefinition to populate the DbType. What I am doing now is that the function will remain of return type boolean but I have moved to mssql specific logic to the mssql class.